base/server/src/main/java/org/dogtagpki/server/rest/v2/JobServletBase.java

```diff
@@ -0,0 +1,224 @@
+//
+// Copyright Red Hat, Inc.
+//
+// SPDX-License-Identifier: GPL-2.0-or-later
+//
+package org.dogtagpki.server.rest.v2;
+
+import java.io.PrintWriter;
+import java.security.Principal;
+import java.util.Arrays;
+import java.util.Enumeration;
+import java.util.List;
+import java.util.Map;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+
+import org.dogtagpki.job.JobCollection;
+import org.dogtagpki.job.JobInfo;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.netscape.certsrv.base.EBaseException;
+import com.netscape.certsrv.base.ForbiddenException;
+import com.netscape.certsrv.base.ResourceNotFoundException;
+import com.netscape.cms.realm.PKIPrincipal;
+import com.netscape.cmscore.apps.CMSEngine;
+import com.netscape.cmscore.apps.EngineConfig;
+import com.netscape.cmscore.jobs.JobConfig;
+import com.netscape.cmscore.jobs.JobsConfig;
+import com.netscape.cmscore.jobs.JobsScheduler;
+import com.netscape.cmscore.jobs.JobsSchedulerConfig;
+
+/**
+ * @author Marco Fargetta {@literal <mfargett@redhat.com>}
+ * @author Endi S. Dewata
+ */
+public class JobServletBase {
+    public static final Logger logger = LoggerFactory.getLogger(JobServletBase.class);
+
+    private CMSEngine engine;
+
+    public JobServletBase(CMSEngine engine) {
+        this.engine = engine;
+    }
+
+    public void get(HttpServletRequest request, HttpServletResponse response) throws Exception {
+        HttpSession session = request.getSession();
+        logger.debug("JobServletBase.get(): session: {}", session.getId());
+
+        PrintWriter out = response.getWriter();
+        if (request.getPathInfo() == null) {
+            JobCollection jobs = findJobs(request.getUserPrincipal());
+            out.println(jobs.toJSON());
+            return;
+        }
+        String[] pathElement = request.getPathInfo().substring(1).split("/");
+        if (pathElement.length == 1) {
+            String jobId = pathElement[0];
+            JobInfo job = getJob(jobId, request.getUserPrincipal());
+            out.println(job.toJSON());
+            return;
+        }
+        response.setStatus(HttpServletResponse.SC_NOT_FOUND);
+    }
+
+    public void post(HttpServletRequest request, HttpServletResponse response) throws Exception {
+        HttpSession session = request.getSession();
+        logger.debug("SelfTestServletBase.post(): session: {}", session.getId());
+
+        if (request.getPathInfo() == null) {
+            response.setStatus(HttpServletResponse.SC_NOT_FOUND);
+            return;
+        }
+        String[] pathElement = request.getPathInfo().substring(1).split("/");
+        if (pathElement.length == 2 && pathElement[1].equals("start")) {
+            String jobId = pathElement[0];
+            startJob(jobId, request.getUserPrincipal());
+            return;
+        }
+        response.setStatus(HttpServletResponse.SC_NOT_FOUND);
```

Comment:
Is it possible to do something like this?
```
if (path length == 1) {
    return method not allowed
} else if (path length == 2) {
    if (element == "start") {
        start job
    } else {
        return not found
    }
} else {
    return not found
}
```

Reply:
This has the problem to verify if the path with length == 1 is handled by other methods. If we add a new path to a method all the other need to be updated.